### ISSUE TITLE
Cancel subscription error after upgrade from 10.5 to 12.8

### ISSUE THREAD

**Issue by roman-briazgalov** [laravel/cashier-stripe#1125]
- Cashier Version: 12.10.0
- Laravel Version: 8.33.1
- PHP Version: 7.4.16
- Database Driver & Version: Mysql5.7

### Description:
I used Laravel framework, Cashier, MongoDB and Jenssegers/laravel-mongodb. But after [this reply](https://github.com/laravel/cashier-stripe/issues/1108#issuecomment-811926970) i have changed database to MySQL for tables users, subscriptions and subscription_items.
My app was working well. But after i have updated Laravel Framework from 6 to 8 and Cashier (stripe) from 10 to 12 cancellation of the subscription stopped working.
I have error:

> Stripe\Exception\UnexpectedValueException
> Could not determine which URL to request: Stripe\Subscription instance has invalid ID:

It looks like the Cashier wants to work with model Subscription of old type (where the field 'stripe_id' existed and wasn't empty, all information about subscription is saved in one table 'subscription'). But the new subscription was made like the multiple subscription, i mean information about subscription is saved in two tables: 'subscriptions' and 'subscription_items'.

If we check the trace of error, we see that here 'vendor/stripe/stripe-php/lib/ApiResource.php:104' is the problem:

```
public function instanceUrl()
{
    return static::resourceUrl($this['id']);
}
```
$this['id'] return null, because there is no 'stripe_id' in subscription object. It has to use table subscription_items and stripe ID inside this table.

I have doublechecked upgrade guides for Laravel and Cashier but could not find anything that could help me.

Could you please show where is the mistake in my part or fix the problem in the lib?

**Comment by driesvints** [comment]
Hi there,

Thanks for reporting but it looks like this is a question which can be asked on a support channel. Please only use this issue tracker for reporting bugs with the library itself. If you have a question on how to use functionality provided by this repo you can try one of the following channels:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

However, this issue will not be locked and everyone is still free to discuss solutions to your problem!

Thanks.

**Comment by roman-briazgalov** [comment]
Hi, @driesvints 

Thanks for your reply!
I hope it could be here because Cashier stopped working properly after upgrade. I decided to post it here exactly because of this.

Let me explain in details:
- my app was working properly;
- i have updated Laravel Framework and Laravel Cashier by official guide;
- subscriptions cancelation stopped working.

So i have decided that it could be bug.

**Comment by roman-briazgalov** [comment]
I have moticed one more weird thing:
In Laravel Cashier docs we see that if we use the second argument in the newSubscription() as the string, we have to get the usual subscription:
```
$request->user()->newSubscription(
    'default', 'price_premium'
)->create($request->paymentMethodId);
```

If we use the second argument as the array, we will get the multiple subscription:
```
$request->user()->newSubscription('default', [
    'price_monthly',
    'chat-plan',
])->create($request->paymentMethodId);
```

I have checked my code. I use string, but in result i have the multiple subscription. It is not OK, right?
